test/ad-cue-tags.test.js

```diff
@@ -0,0 +1,292 @@
+        cueOutCont: '20/30'
+      },
+      {
+        duration: 10,
+        cueIn: ''
+      }
+    ]
+  }, this.track, 30);
+
+  QUnit.equal(this.track.cues.length, 1, 'did not remove cue or add a new one');
+
+  QUnit.equal(testCue.startTime, 10, 'cue still starts at 10');
+  QUnit.equal(testCue.endTime, 40, 'cue end still 40');
+  QUnit.equal(testCue.adStartTime, 10, 'cue ad still starts at 10');
+  QUnit.equal(testCue.adEndTime, 40, 'cue ad still ends at 40');
+});
+
+QUnit.test('adjust cue end time in event of early CUE-IN', function() {
+  AdCueTags.updateAdCues({
+    segments: [
+      {
+        duration: 10,
+        cueOut: '30'
+      },
+      {
+        duration: 10,
+        cueOutCont: '10/30'
+      },
+      {
+        duration: 10,
+        cueOutCont: '20/30'
+      }
+    ]
+  }, this.track, 10);
+
+  QUnit.equal(this.track.cues.length, 1, 'adds a single cue for new ad');
+
+  let testCue = this.track.cues[0];
+
+  QUnit.equal(testCue.startTime, 10, 'cue starts at 10');
+  QUnit.equal(testCue.endTime, 40, 'cue ends at start time plus segment durations');
+  QUnit.equal(testCue.adStartTime, 10, 'cue ad starts at 10');
+  QUnit.equal(testCue.adEndTime, 40, 'cue ad ends at 40');
+
+  AdCueTags.updateAdCues({
+    segments: [
+      {
+        duration: 10,
+        cueOutCont: '10/30'
+      },
+      {
+        duration: 10,
+        cueIn: ''
+      },
+      {
+        duration: 10
+      }
+    ]
+  }, this.track, 20);
+
+  QUnit.equal(this.track.cues.length, 1, 'did not remove cue or add a new one');
+
+  QUnit.equal(testCue.startTime, 10, 'cue still starts at 10');
+  QUnit.equal(testCue.endTime, 30, 'cue end updated to 30');
+  QUnit.equal(testCue.adStartTime, 10, 'cue ad still starts at 10');
+  QUnit.equal(testCue.adEndTime, 30,
+    'cue ad end updated to 30 to account for early cueIn');
+});
+
+QUnit.test('correctly handle multiple ad cues', function() {
+  AdCueTags.updateAdCues({
+    segments: [
+      {
+        duration: 10
+      },
+      {
+        duration: 10
+      },
+      {
+        duration: 10
+      },
+      {
+        duration: 10,
+        cueOut: '30'
+      },
+      {
+        duration: 10,
+        cueOutCont: '10/30'
+      },
+      {
+        duration: 10,
+        cueOutCont: '20/30'
+      },
+      {
+        duration: 10,
+        cueIn: ''
+      },
+      {
+        duration: 10
+      },
+      {
+        duration: 10
+      },
+      {
+        duration: 10
+      },
+      {
+        duration: 10,
+        cueOut: '20'
+      },
+      {
+        duration: 10,
+        cueOutCont: '10/20'
+      },
+      {
+        duration: 10,
+        cueIn: ''
+      },
+      {
+        duration: 10
+      }
+    ]
+  }, this.track);
+
+  QUnit.equal(this.track.cues.length, 2, 'correctly created 2 cues for the ads');
+  QUnit.equal(this.track.cues[0].startTime, 30, 'cue created at correct start time');
+  QUnit.equal(this.track.cues[0].endTime, 60, 'cue created at correct end time');
```

Comment:
Minor note, but phrasing for end times seems a bit off

Also, while they should be the same, can't hurt to check adStartTimes and adEndTimes as well